src/lib/consensus/proof_of_stake.ml

```diff
@@ -1013,99 +1027,105 @@ module Make (Inputs : Inputs_intf) : Intf.S = struct
       ; last_epoch_data
       ; curr_epoch_data }
 
-    let update_var (previous_state : var)
-        (transition_data : Consensus_transition_data.var)
-        (previous_protocol_state_hash : Coda_base.State_hash.var)
-        ~(supply_increase : Currency.Amount.var)
-        ~(previous_blockchain_state_ledger_hash :
-           Coda_base.Frozen_ledger_hash.var) =
-      let open Snark_params.Tick in
-      let open Let_syntax in
-      let prev_epoch = previous_state.curr_epoch in
-      let next_epoch = transition_data.epoch in
-      let next_slot = transition_data.slot in
-      let%bind epoch_increased =
-        let%bind c = Epoch.compare_var prev_epoch next_epoch in
-        let%map () = Boolean.Assert.is_true c.less_or_equal in
-        c.less
-      in
-      let%bind last_data =
-        Epoch_data.if_ epoch_increased ~then_:previous_state.curr_epoch_data
-          ~else_:previous_state.last_epoch_data
-      in
-      let%bind threshold_satisfied, vrf_result =
-        let%bind (module M) = Inner_curve.Checked.Shifted.create () in
-        Vrf.Checked.check
-          (module M)
-          ~epoch_ledger:last_data.ledger ~epoch:transition_data.epoch
-          ~slot:transition_data.slot ~seed:last_data.seed
-      in
-      let%bind curr_data =
-        let%map seed =
-          let%bind in_seed_update_range =
-            Epoch.Slot.in_seed_update_range_var next_slot
-          in
-          let%bind base =
-            Epoch_seed.if_ epoch_increased
-              ~then_:Epoch_seed.(var_of_t initial)
-              ~else_:previous_state.curr_epoch_data.seed
-          in
-          let%bind updated = Epoch_seed.update_var base vrf_result in
-          Epoch_seed.if_ in_seed_update_range ~then_:updated ~else_:base
-        and length =
-          let%bind base =
-            Field.Checked.if_ epoch_increased
-              ~then_:Field.(Checked.constant zero)
-              ~else_:
-                ( Length.pack_var previous_state.curr_epoch_data.length
-                  :> Field.var )
-          in
-          Length.var_of_field Field.(Checked.(add (constant one) base))
-        and ledger =
-          Epoch_ledger.if_ epoch_increased
-            ~then_:
-              { total_currency= previous_state.total_currency
-              ; hash= previous_blockchain_state_ledger_hash }
-            ~else_:previous_state.curr_epoch_data.ledger
-        and start_checkpoint =
-          Coda_base.State_hash.if_ epoch_increased
-            ~then_:previous_protocol_state_hash
-            ~else_:previous_state.curr_epoch_data.start_checkpoint
-        (* Want this to be the protocol state hash once we leave the seed
-           update range. *)
-        and lock_checkpoint =
-          let%bind base =
-            (* TODO: Should this be zero or some other sentinel value? *)
+    module Scoped = struct
```

Comment:
`let%snarkydef` doesn't work unless you `include Tick`, but `include Tick` here shadows something that causes a compile error later.

I meant for these to just be temporary, but I wonder if we should keep them and just do something better with these inner modules

Reply:
I would call that a bug in `let%snarkydef` then that we should fix. But you don't actually call `include Tick` in these scoped modules, you only `open Crypto_params_init.Tick0`. So is this still really necessary?

Reply:
yes by `include Tick` I mean `open Crypto_params_init.Tick0`, and I agree it's a bug in `let%snarkydef`: see https://github.com/o1-labs/snarky/issues/40 . This was very helpful for debugging so it would be nice if we could keep the `let%snarkydef`s -- I can do `module M = struct end; include M` if that's better?

Reply:
@nholland94  said I can use include struct